samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -84,6 +84,7 @@ private String getHostName() {
 
   @Override
   public void tryBecomeLeader(LeaderElectorListener leaderElectorListener) {
+    zkUtils.makeSurePersistentPathsExists(new String[]{keyBuilder.getProcessorsPath()});
```

Comment:
Why this line added here?

This was previously done in the constructor. Line 71-73 are commented out. What is the reason for this change?